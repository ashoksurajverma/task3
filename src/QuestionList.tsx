import React from "react";
import QuestionContainer from "./QuestionContainer";

interface IQuestion {
  question: string;
  answer: string;
}

interface IProps {
  questions: Array<IQuestion>;
}
function QuestionList({ questions }: IProps) {
  return (
    <div className="question-wrapper">
      {questions.map((question: IQuestion, index: number) => {
        return <QuestionContainer {...question} key={index} />;
      })}
    </div>
  );
}

export default QuestionList;
