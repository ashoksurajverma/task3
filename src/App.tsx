import React, { useState } from "react";
import QuestionList from "./QuestionList";
import "./app.css";

function App() {
  const [questions, setQuestions] = useState([
    {
      question:
        "What is the average the airspeed velocity of a (European) unladen swallow?",
      answer: "11 meters per second",
    },
    {
      question:
        "What is the average the airspeed velocity of a (European) unladen swallow?",
      answer: "11 meters per second",
    },
    {
      question:
        "What is the average the airspeed velocity of a (European) unladen swallow?",
      answer: "11 meters per second",
    },
    {
      question:
        "What is the average the airspeed velocity of a (European) unladen swallow?",
      answer: "11 meters per second",
    },
    {
      question:
        "What is the average the airspeed velocity of a (European) unladen swallow?",
      answer: "11 meters per second",
    },
    {
      question:
        "What is the average the airspeed velocity of a (European) unladen swallow?",
      answer: "11 meters per second",
    },
  ]);

  return (
    <div id="app">
      <QuestionList questions={questions} />
    </div>
  );
}

export default App;
