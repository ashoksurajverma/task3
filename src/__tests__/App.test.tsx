import * as React from "react";
import { shallow, ShallowWrapper, configure } from "enzyme";
// @ts-ignore
import Adapter from "enzyme-adapter-react-16";

configure({ adapter: new Adapter() });
import App from "../App";
import QuestionList from "../QuestionList";

describe("Render App component", () => {
  let wrapper: ShallowWrapper<undefined, undefined>;
  beforeEach(() => (wrapper = shallow(<App />)));

  it("should render without error", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should have one children", () => {
    expect(wrapper.children().find(QuestionList).length).toBe(1);
  });
});
