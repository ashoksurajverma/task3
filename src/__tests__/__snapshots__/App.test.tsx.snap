// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render App component should render without error 1`] = `ShallowWrapper {}`;
